### PR TITLE
[codegen/nodejs] Refactor GeneratePackage method

### PR DESCRIPTION
**Pull request by lblackstone**
- Split out common functionality into generateModuleContextMap method
- Add LanguageResource and LanguageProperty structs for use by downstream
codegen tools
- Add LanguageResources function for use by downstream codegen tools

Related to https://github.com/pulumi/pulumi-kubernetes/issues/1053

### CODE REVIEW

**Comment by lblackstone** [comment]
Azure downstream failure appears to be unrelated. RFR @pgavlin

File: pkg/codegen/nodejs/gen.go [hunk 1]
```diff
@@ -1417,6 +1415,81 @@ func GeneratePackage(tool string, pkg *schema.Package, extraFiles map[string][]b
...
+// LanguageProperty holds information about a resource property to be used by downstream codegen.
+type LanguageProperty struct {
+	ConstValue string // If set, the constant value of the property (e.g., "flowcontrol.apiserver.k8s.io/v1alpha1")
+	Name       string // The name of the property (e.g., "FlowSchemaSpec")
+	Package    string // The package path containing the property definition (e.g., "outputs.flowcontrol.v1alpha1")
+}
+
+// LanguageResources returns a map of resources that can be used by downstream codegen. The map
+// key is the resource schema token.
+func LanguageResources(tool string, pkg *schema.Package) (map[string]LanguageResource, error) {
```
Comment (pgavlin) [inline pkg/codegen/nodejs/gen.go:1439]: Does the `tool` parameter doesn't influence the output? If not, can we remove it (and just pass `""` to `generateModuleContextMap`?

```suggestion
func LanguageResources(pkg *schema.Package) (map[string]LanguageResource, error) {
```

File: pkg/codegen/nodejs/gen.go [hunk 2]
```diff
@@ -1417,6 +1415,81 @@ func GeneratePackage(tool string, pkg *schema.Package, extraFiles map[string][]b
...
+
+	modules, err := generateModuleContextMap(tool, pkg, info)
+	if err != nil {
+		return nil, err
+	}
+
+	for modName, mod := range modules {
+		if modName == "" {
+			continue
+		}
```
Comment (pgavlin) [inline pkg/codegen/nodejs/gen.go:1455]: Why is this necessary? The root module may contain resources, functions, etc.

File: pkg/codegen/nodejs/gen.go (hunk 1 above)
Comment (lblackstone) [reply pkg/codegen/nodejs/gen.go:1439]: Nope, looks like we can remove it.

File: pkg/codegen/nodejs/gen.go (hunk 2 above)
Comment (lblackstone) [reply pkg/codegen/nodejs/gen.go:1455]: I guess it's not required for k8s, but I'll handle this case downstream instead.

**Comment by lblackstone** [comment]
> I assume that the definitions of `LanguageResource` and `LanguageProperty` contain the minimum that's necessary for k8s?

Yes, this is the minimum needed for YAML support.
